src/object.c

```diff
@@ -51,14 +51,39 @@ static git_object_def git_objects_table[] = {
 	{ "REF_DELTA", 0, NULL, NULL },
 };
 
+int git_object__init_with_odb(git_object **object_out, git_odb_object *odb_obj, git_repository *repo)
```

Comment:
I would have loved too 😆. IIRC it's a side-effect of the parse merging. After a more thorough look, it might just be confusion : the revwalk-parsing works on `git_commit`, and the normal parsing did it, so I made them *both* do it. Will investigate if it's really necessary (I remember a crash when freeing though).